src/gui/properties/trackerlist.cpp

```diff
@@ -529,13 +543,61 @@ void TrackerList::showTrackerListMenu(QPoint) {
 
 void TrackerList::loadSettings()
 {
-    if (!header()->restoreState(Preferences::instance()->getPropTrackerListState())) {
-        setColumnWidth(0, 30);
-        setColumnWidth(1, 300);
-    }
+    header()->restoreState(Preferences::instance()->getPropTrackerListState());
 }
 
 void TrackerList::saveSettings() const
 {
     Preferences::instance()->setPropTrackerListState(header()->saveState());
 }
+
+QStringList TrackerList::headerLabels()
+{
+    QStringList header;
+    header << "#";
+    header << tr("URL");
+    header << tr("Status");
+    header << tr("Received");
+    header << tr("Seeds");
+    header << tr("Peers");
+    header << tr("Downloaded");
+    header << tr("Message");
+
+    return header;
+}
+
+void TrackerList::displayToggleColumnsMenu(const QPoint&)
+{
+    QMenu hideshowColumn(this);
+    hideshowColumn.setTitle(tr("Column visibility"));
+    QList<QAction*> actions;
+    for (int i = 0; i < COL_COUNT; ++i) {
+        QAction *myAct = hideshowColumn.addAction(headerLabels().at(i));
+        myAct->setCheckable(true);
+        myAct->setChecked(!isColumnHidden(i));
+        actions.append(myAct);
+    }
+    int visibleCols = 0;
+    for (unsigned int i = 0; i < COL_COUNT; i++) {
```

Comment:
In a later commit it's moved in a seperate method in order to use it in the constructor as well. Take a look and let me know if you have a better idea.